extensions/amp-analytics/0.1/transport.js

```diff
@@ -118,13 +143,262 @@ export class Transport {
     xhr.send('');
     return true;
   }
+
+  /**
+   * If iframe is specified in config/transport, check whether third-party
+   * iframe already exists, and if not, create it.
+   * @param {!HTMLDocument} ampDoc The AMP document
+   * @param {!Object<string,string>} transportOptions The 'transport' portion
+   * of the amp-analytics config object
+   * @param {function(!string,string)=} opt_processResponse An optional
+   * function to receive any response messages back from the cross-domain iframe
+   */
+  processCrossDomainIframe(ampDoc, transportOptions, opt_processResponse) {
+    dev().assert(transportOptions['iframe'],
+      'Cross-domain frame parameters missing');
+    const frameUrl = transportOptions['iframe'];
+    this.beginUsingCrossDomainIframe_(ampDoc, frameUrl,
+      transportOptions['extraData']);
+    const frameData = Transport.crossDomainIframes_[frameUrl];
+    const iframeMessagingClient = frameData.iframeMessagingClient;
+    iframeMessagingClient.registerCallback(
+      'ampAnalytics3pReady', () => {
+        iframeMessagingClient.setHostWindow(frameData.frame.contentWindow);
+        Transport.setIsReady_(frameUrl);
+      });
+    if (!opt_processResponse) {
+      return;
+    }
+    iframeMessagingClient.registerCallback(
+      'ampAnalytics3pResponse', response => {
+        if (!response || !response.ampAnalytics3pResponse) {
+          return;
+        }
+        opt_processResponse(this.type_, response.ampAnalytics3pResponse);
```

Comment:
Consider removing opt_processResponse existence check and instead put it here such that if you receive a message you did not expect, you do dev().warn('unexpected response...') so that developers can get feedback as to why integration is not working